### PR TITLE
[TASKMGR] CachedGetUserFromSid(): Properly support NULL 'pSid'

### PR DESCRIPTION
**Pull request by SergeGautherie**
With regard to `SystemUserSid == NULL`...

### CODE REVIEW

**Comment by Extravert-ir** [comment]
Can you reproduce the problem while using the app?

**Comment by SergeGautherie** [comment]
> Can you reproduce the problem while using the app?

On LiveCD,

RealLife:
I don't know how to trigger this edge case.
All processes are displayed with `Username == 'SYSTEM'`.

Test:
Skip `AllocateAndInitializeSid()` call.
'System' and 'System Idle Process' processes use new code path and are displayed with empty 'Username'.

File: base/applications/taskmgr/perfdata.c [hunk 1]
```diff
@@ -125,14 +128,22 @@ static void SidToUserName(PSID Sid, LPWSTR szBuffer, DWORD BufferSize)
 VOID
 WINAPI
 CachedGetUserFromSid(
```
Comment (ThFabba) [inline base/applications/taskmgr/perfdata.c:130]: On Windows, `CachedGetUserFromSid` is a function exported from `utildll.dll`. Can you write a test that shows how it behaves with a NULL Sid? Thanks.

File: base/applications/taskmgr/perfdata.c [hunk 2]
```diff
@@ -125,14 +128,22 @@ static void SidToUserName(PSID Sid, LPWSTR szBuffer, DWORD BufferSize)
...
+    _Out_ LPWSTR pUserName,
+    _Inout_ PULONG pcwcUserName)
 {
     PLIST_ENTRY pCur;
     PSIDTOUSERNAME pEntry;
     ULONG cbSid, cwcUserName;
 
+    if (!pSid)
+    {
+        DPRINT("No pSid\n");
```
Comment (ThFabba) [inline base/applications/taskmgr/perfdata.c:141]: The only way for this to become null is if `PerfDataInitialize` fails to allocate memory. So it would make more sense to print there (once) than here (every loop).

File: base/applications/taskmgr/perfdata.c (hunk 1 above)
Comment (SergeGautherie) [reply base/applications/taskmgr/perfdata.c:130]: Thanks for the information.
Done: #2954.

File: base/applications/taskmgr/perfdata.c (hunk 2 above)
Comment (SergeGautherie) [reply base/applications/taskmgr/perfdata.c:141]: `PerfDataInitialize()` is (already) taken care of on #2946.
`GetSecurityInfo()` documentation says it could fail too: I'll move this trace there...

File: base/applications/taskmgr/perfdata.c (hunk 1 above)
Comment (SergeGautherie) [reply base/applications/taskmgr/perfdata.c:130]: > `utildll.dll`

Shall we also start creating this dll? And plan to move `CachedGetUserFromSid()` from here there?

**Comment by Extravert-ir** [comment]
Too much PRs for a single module
